### PR TITLE
Drupal Toolbox Extension

### PR DESCRIPTION
**Pull request by emircanerkul**
## Description

New Extension: Drupal Toolbox extension

Valuable tools and shortcuts for your Drupal Websites. Manage your Drupal Websites with ease.

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/drupal-toolbox/src/tools.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    id: "open_in_php_storm",
+    title: "Open in PhpStorm",
+    action: (website: DrupalWebsite) =>
+      Tool(website, `phpstorm ${website.root}`, {
+        inprogress: "Preparing to Open in PhpStorm.",
+        completed: "The project opened in PhpStorm.",
+      }),
+    icon: Icon.CodeBlock,
+  },
+];
```
Comment (pernielsentikaer) [inline extensions/drupal-toolbox/src/tools.tsx:68]: Could we add [ActionPanel](https://developers.raycast.com/api-reference/user-interface/actions#api-reference) icons here

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have a suggestions ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/drupal-toolbox/src/tools.tsx (hunk 1 above)
Comment (emircanerkul) [reply extensions/drupal-toolbox/src/tools.tsx:68]: Also, thank you for Destructive Style

**Comment by pernielsentikaer** [comment]
Hi @emircanerkul 👋 

I tested it once more, and I don't see a reason to push to another view where only actions are possible, what do you think about doing this instead?

drupal-toolbox 2022-11-30 at 06 02 23

**Comment by emircanerkul** [comment]
> Hi @emircanerkul 👋
> 
> I tested it once more, and I don't see a reason to push to another view where only actions are possible, what do you think about doing this instead?
> 
> drupal-toolbox 2022-11-30 at 06 02 23

To be able to filter, to be able to extend without thinking how many tools have in the list (I think it's not so much usable in that little action menu), and for future possibilities like nested tools.

**Comment by pernielsentikaer** [comment]
Hi @emircanerkul 👋 

That sounds like a good idea to keep tools then, thanks for the clarification 🔥 

I edited a few things, I hope it's okay with you 😊

- Use useForm for adding the sites, this removed a lot of code
- Added `canChooseFiles={false}` to the picker, so only folders can be selected
- Updated API
- Small wording touches

As always, take a look and let me know, everything is up for debate 😊

File: extensions/drupal-toolbox/src/components/EmptyView.tsx [hunk 2]
```diff
@@ -25,8 +25,8 @@ function EmptyView(props: {
   return (
     <List.EmptyView
       icon="drupal-toolbox.png"
-      title="No Drupal websites found"
-      description="You don't have any Drupal websites yet. Why not add some?"
+      title="No Sites Found"
+      description="Press ⌘+N to add a site"
```
Comment (emircanerkul) [inline extensions/drupal-toolbox/src/components/EmptyView.tsx:29]: Drupal, drupal, drupal, druuuupal: https://youtu.be/lZ-s3DRZJKY?t=11

Comment (emircanerkul) [reply extensions/drupal-toolbox/src/components/EmptyView.tsx:29]: But okay, not a big deal. I'll stop being drupal partisan :)

**Comment by emircanerkul** [comment]
> Hi @emircanerkul 👋
> 
> That sounds like a good idea to keep tools then, thanks for the clarification 🔥
> 
> I edited a few things, I hope it's okay with you 😊
> 
> * Use useForm for adding the sites, this removed a lot of code
> * Added `canChooseFiles={false}` to the picker, so only folders can be selected
> * Updated API
> * Small wording touches
> 
> As always, take a look and let me know, everything is up for debate 😊

All is okay except for one point that is;

I was already placed canChooseDirectories={true} I'm not sure also canChooseFiles={false} is required. By default, it should not if I mention it as a directory, but not sure if it's a good design. Might be enum could be good for this case? type={Form.FilePicker.Type.File|Form.FilePicker.Type.Folder|Form.FilePicker.Type.Both}

Also good to have this validation section in the example todo app. I just based on this :)

**Comment by pernielsentikaer** [comment]
> I was already placed canChooseDirectories={true} I'm not sure also canChooseFiles={false} is required. By default, it should not if I mention it as a directory, but not sure if it's a good design. Might be enum could be good for this case? type={Form.FilePicker.Type.File|Form.FilePicker.Type.Folder|Form.FilePicker.Type.Both}

If you don't set `canChooseFiles={false}` then you can both pick a file AND a folder since it defaults on

Yes, our new useForm is much better since it handles all the scenarios with validations and remembering data

File: extensions/drupal-toolbox/src/components/EmptyView.tsx (hunk 2 above)
Comment (pernielsentikaer) [reply extensions/drupal-toolbox/src/components/EmptyView.tsx:29]: It's just to shorten the text for the user and this way makes the UI/UX a bit easier. Damn, that's a catchy song 😊

Comment (emircanerkul) [reply extensions/drupal-toolbox/src/components/EmptyView.tsx:29]: Happy to your mind :)

**Comment by emircanerkul** [comment]
I think we are all done, right?

**Comment by pernielsentikaer** [comment]
We should be 🚀

**Comment by pernielsentikaer** [comment]
Thanks for your cooperation, it was nice working with you 🔥

**Pull request merged**